backend/src/modules/users/users.service.ts

```diff
@@ -1,33 +1,21 @@
-import { BadRequestException, Injectable } from '@nestjs/common';
+import { HttpCode, HttpStatus, Injectable } from '@nestjs/common';
 import { CreateUserDto } from './dto/create-users.dto';
 import * as argon2 from 'argon2';
-import { UserRepository } from './users.repository';
+import { UsersRepository } from './users.repository';
 
 @Injectable()
 export class UsersService {
-  constructor(private readonly userRepository: UserRepository) {}
+  constructor(private readonly userRepository: UsersRepository) {}
 
+  @HttpCode(HttpStatus.CREATED)
   async create(createUserDto: CreateUserDto) {
-    const existUser = await this.userRepository.findOne({
```

Comment:
We need this validation, because we want to throw error on the highest lvl of the app, so in this case we should throw error on the service layer and don't provide invalid data to database (because unique constraint will throw error in the database layer)